### ISSUE TITLE
CLI: Option to skip file output if the same file already exists

### ISSUE THREAD

**Issue by natew** [babel/babel#6445]
Consider this a small feature request, or perhaps even something solvable in shell (though I know not how).

Basically, when doing an `--out-dir` with the CLI, it would be helpful to have a flag `--skip-write-if-same`. Reason being, if you have watchers running like webpack, it will get pretty frustrated by all the file writes as you run watchers.

I figure a PR wouldn't be too tricky to do, so if people think this is a good idea I could give it a go.

**Comment by xtuc** [comment]
I think this should actually be the default. There is no reason to re-compile if the source file hasn't changed but there might be a few edge cases I don't have in mind currently.

**Comment by loganfsmyth** [comment]
Interesting idea, I guess that would be relatively easily to do. I'm up for accepting a PR for this I think.

@xtuc I don't think we can avoid recompiling at the moment since we can't easily know if something changed, but we can definitely recompile and then just skip writing to disk if the content is already there.

**Comment by hzoo** [comment]
The edge case is that the source is the same, but you changed your babelrc/node_modules and babel version(s) unless this is only during "watch"?

**Comment by xtuc** [comment]
Sorry, I wasn't clear, I meant to avoid Babel writing a file with the same content which triggers a build in some watcher.

On the other hand we could store hashes to detect if a file has changed in Babel's input.

**Comment by loganfsmyth** [comment]
Yeah, storing hashes and such is where things start getting complicated. I have idea for that hopefully for 7.x, but it seems beyond the scope of what this is asking for.

**Comment by xtuc** [comment]
Yes, but as @hzoo said, a checksum of the content of the file in input won't suffice, it's not that easy.

About the output, it's simpler. I would compare the emited string from Babel with the current content of the file before writing in it.

**Comment by loganfsmyth** [comment]
Agreed.

**Comment by natew** [comment]
Actually, there's a much simpler option that would actually solve my use case at least.

`--watch-incoming`

Basically: Starts a watch but doesn't compile anything until it sees a new change. That would let me then use `babel-changed` to do an initial compile (or just run a full compile way less often), and then I could run babel `--watch-incoming` without it re-building everything.

Currently working in a mono-repo that is growing, and starting to feel the pain. With 3 different compile targets * 26 packages, it starts to get a bit painful.

If this sounds appealing I'd happily send PR, seems like it should be fairly simple.

**Comment by natew** [comment]
Holy crap this already exists (re: last comment)... `--skip-initial-build`

**Comment by rvion** [comment]
@natew  it doesn't seem to be the same.

skip-initial-build also skip build of files that have changed since. it's a dangerous flag for most people since it can provoque hard to debug problems.

a `--skip-write-if-same` would very much make sense IMO, big `+1`
